lib/db/mysql.js

```diff
@@ -1451,33 +1456,36 @@ module.exports = function (log, error) {
   }
 
   const DELETE_RECOVERY_CODES = 'CALL deleteRecoveryCodes_1(?)'
-  const INSERT_RECOVERY_CODE = 'CALL createRecoveryCode_1(?, ?)'
+  const INSERT_RECOVERY_CODE = 'CALL createRecoveryCode_2(?, ?, ?)'
   MySql.prototype.replaceRecoveryCodes = function (uid, count) {
 
     // Because of the hashing requirements the process of replacing
     // recovery codes is done is two separate procedures. First one
     // deletes all current codes and the second one inserts the
     // hashed randomly generated codes.
-    return dbUtil.generateRecoveryCodes(count)
-      .then((codeList) => {
+    return dbUtil.generateRecoveryCodes(count, RECOVERY_CODE_KEYSPACE, RECOVERY_CODE_LENGTH)
+      .then((codes) => {
         return this.read(DELETE_RECOVERY_CODES, [uid])
           .then(() => {
-            if (codeList <= 0) {
+            if (codes <= 0) {
```

Comment:
I realise it was here before (and I probably reviewed it before too) but this line looks funny to me now, unless I'm missing something. Isn't `codes` guaranteed to be an array? When would this test ever succeed?

Maybe it was meant to be a comparison to `codes.length` but even then it seems unnecessary because `codes.map` will be a nop for the empty array case.

Reply:
Oh right, it's one of those JS "wat" moments:

```
> [] <= 0
true
> [] === 0
false
> [] == 0
true
```

Seems a bit esoteric to me!

Reply:
Whoops it was mean to be `codes.length`, but like you said isn't needed. Removed!